crates/democracy/src/lib.rs

```diff
@@ -0,0 +1,1154 @@
+        Self::launch_public(now).map_err(|_| Error::<T>::NoneWaiting.into())
+    }
+
+    /// Table the waiting public proposal with the highest backing for a vote.
+    fn launch_public(now: T::BlockNumber) -> DispatchResult {
+        let mut public_props = Self::public_props();
+        if let Some((winner_index, _)) = public_props.iter().enumerate().max_by_key(
+            // defensive only: All current public proposals have an amount locked
+            |x| Self::backing_for((x.1).0).unwrap_or_else(Zero::zero),
+        ) {
+            let (prop_index, proposal, _) = public_props.swap_remove(winner_index);
+            <PublicProps<T>>::put(public_props);
+
+            if let Some((depositors, deposit)) = <DepositOf<T>>::take(prop_index) {
+                // refund depositors
+                for d in &depositors {
+                    T::Currency::unreserve(d, deposit);
+                }
+                Self::deposit_event(Event::<T>::Tabled(prop_index, deposit, depositors));
+                Self::inject_referendum(
+                    now + T::VotingPeriod::get(),
+                    proposal,
+                    VoteThreshold::SuperMajorityAgainst,
+                    T::EnactmentPeriod::get(),
+                );
+            }
+            Ok(())
+        } else {
+            Err(Error::<T>::NoneWaiting)?
+        }
+    }
+
+    fn do_enact_proposal(proposal_hash: T::Hash, index: ReferendumIndex) -> DispatchResult {
+        let preimage = <Preimages<T>>::take(&proposal_hash);
+        if let Some(PreimageStatus::Available {
+            data,
+            provider,
+            deposit,
+            ..
+        }) = preimage
+        {
+            if let Ok(proposal) = T::Proposal::decode(&mut &data[..]) {
+                let err_amount = T::Currency::unreserve(&provider, deposit);
+                debug_assert!(err_amount.is_zero());
+                Self::deposit_event(Event::<T>::PreimageUsed(proposal_hash, provider, deposit));
+
+                let res = proposal
+                    .dispatch(frame_system::RawOrigin::Root.into())
+                    .map(|_| ())
+                    .map_err(|e| e.error);
+                Self::deposit_event(Event::<T>::Executed(index, res));
+
+                Ok(())
+            } else {
+                T::Slash::on_unbalanced(T::Currency::slash_reserved(&provider, deposit).0);
+                Self::deposit_event(Event::<T>::PreimageInvalid(proposal_hash, index));
+                Err(Error::<T>::PreimageInvalid.into())
+            }
+        } else {
+            Self::deposit_event(Event::<T>::PreimageMissing(proposal_hash, index));
+            Err(Error::<T>::PreimageMissing.into())
+        }
+    }
+
+    fn bake_referendum(
+        now: T::BlockNumber,
+        index: ReferendumIndex,
+        status: ReferendumStatus<T::BlockNumber, T::Hash, BalanceOf<T>>,
+    ) -> Result<bool, DispatchError> {
+        let total_issuance = T::Currency::total_issuance();
```

Comment:
Not sure how this exactly will work if voting power decreases over time

Reply:
Good point, when I integrate the escrow pallet it will probably have to dynamically calculate the tally from the current block height.